src/allmydata/test/cli/test_start.py

```diff
@@ -254,7 +254,11 @@ def stop(*args, **kw):
         ])
 
         i, o, e = StringIO(), StringIO(), StringIO()
-        runner.dispatch(config, i, o, e)
+        d = runner.dispatch(config, i, o, e)
+
+        def must_be_systemexit(f):
+            assert isinstance(f.value, SystemExit)
+        d.addErrback(must_be_systemexit)
```

Comment:
Probably `self.assertFailure(d, SystemExit)`?

https://twistedmatrix.com/documents/current/api/twisted.trial.unittest.TestCase.html#assertFailure

Since the old code failed if d didn't errback but as written this new code will let success pass.